src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joined = '';
+
+    for (const item of this) {
+      if (item === null || typeof item === 'undefined') {
+        joined += ',';
+        continue;
+      }
```

Comment:
```suggestion
      if (item !== null || typeof item !== 'undefined') {
         добавляем элемент в строку.
      }
```
добавить еще одну проверку if (i !== this.length - 1) добавляем разделитель в строку.